pyro/infer/mcmc/nuts.py

```diff
@@ -150,24 +155,28 @@ def _build_basetree(self, z, r, z_grads, log_slice, direction, energy_current):
             z, r, self._potential_energy, self._inverse_mass_matrix, step_size, z_grads=z_grads)
         r_new_flat = torch.cat([r_new[site_name].reshape(-1) for site_name in sorted(r_new)])
         energy_new = potential_energy + self._kinetic_energy(r_new)
-        sliced_energy = energy_new + log_slice
 
-        # As a part of the slice sampling process (see below), along the trajectory
-        #     we eliminate states which p(z, r) < u, or dE > 0.
-        # Due to this elimination (and stop doubling conditions),
-        #     the size of binary tree might not equal to 2^tree_depth.
-        tree_size = 1 if sliced_energy <= 0 else 0
```

Comment:
We should modify or remove the comment below now, since we aren't handling this as a special case.